hearth/lib/hearth/http/client.rb

```diff
@@ -81,28 +82,56 @@ def initialize(options = {})
       # @param [Response] response
       # @return [Response]
       def transmit(request:, response:, **options)
-        http = create_http(request.uri)
-        http.set_debug_output(options[:logger]) if @debug_output
-        configure_timeouts(http)
-
-        if request.uri.scheme == 'https'
-          configure_ssl(http)
-        else
-          http.use_ssl = false
+        # http = create_http(request.uri)
+        # http.set_debug_output(options[:logger]) if @debug_output
+        # configure_timeouts(http)
+        #
+        # if request.uri.scheme == 'https'
+        #   configure_ssl(http)
+        # else
+        #   http.use_ssl = false
+        # end
+        endpoint = request.uri
+        pool = ConnectionPool.for(config)
```

Comment:
Rather than getting this pool each time we transmit, the `config` is per instance of the http client right?  Why not get the `pool` once?  That avoids getting two seperate mutex for each transmit.

Reply:
That could work